File: frontend/src/types/analysis.ts
export type ConfidenceLevel = 'HIGH' | 'MEDIUM' | 'LOW' | 'UNKNOWN';

export interface ImageMeta {
  width: number;
  height: number;
  format: string;
}

export interface SpeciesInfo {
  common_name: string;
  scientific_name: string;
  confidence: number;
  confidence_level: ConfidenceLevel;
}

export interface DiseaseInfo {
  name: string;
  scientific_name?: string;
  confidence: number;
  confidence_level: ConfidenceLevel;
  severity?: string;
  symptoms: string[];
  management: string[];
}

export interface ExplainabilityInfo {
  gradcam_url?: string;
  top_regions: string[];
}

export interface BoundingBox {
  x_min: number;
  y_min: number;
  x_max: number;
  y_max: number;
}

export interface DetectionResult {
  category: string;
  species?: SpeciesInfo;
  disease?: DiseaseInfo;
  explainability?: ExplainabilityInfo;
  bounding_box?: BoundingBox;
}

export interface SimilarImage {
  id: string;
  similarity: number;
  thumbnail_url: string;
}

export interface AnalysisResponse {
  id: string;
  timestamp: string;
  image_size: ImageMeta;
  scene_category: string;
  scene_confidence: number;
  detections: DetectionResult[];
  similar_images: SimilarImage[];
  processing_time_ms: number;
  model_versions: Record<string, string>;
}

File: backend/app/schemas/analysis.py
from pydantic import BaseModel
from typing import Optional, List, Dict
from datetime import datetime
from .common import ImageMeta
from .detection import DetectionResult

class SimilarImage(BaseModel):
    id: str
    similarity: float
    thumbnail_url: str

class AnalysisRequest(BaseModel):
    pass  # Usually handled via form-data for file uploads

class AnalysisResponse(BaseModel):
    id: str
    timestamp: datetime
    image_size: ImageMeta
    scene_category: str
    scene_confidence: float
    detections: List[DetectionResult]
    similar_images: List[SimilarImage]
    processing_time_ms: int
    model_versions: Dict[str, str]
